### PR TITLE
feature: crossover point widget to add inflation adjustment

### PR DESCRIPTION
**Pull request by math29**
## Summary by CodeRabbit

* **New Features**
  * Add an annual "Inflation rate" input (percentage) to Crossover Point reports with tooltip; value is saved with the widget and applied to projections using monthly compounding, updating projected expenses, crossover charts, and summary metrics (years to retire, target nest egg).

* **Tests**
  * Add comprehensive tests covering annual→monthly conversion, compound monthly projections, zero/null/negative rates, precision over long horizons, and multiple projection modes.

* **Documentation**
  * Add release-note entry describing the inflation adjustment feature.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

<hr />

### Bundle Stats

Bundle | Files count | Total bundle size | % Changed
------ | ----------- | ----------------- | ---------
desktop-client | 27 | 14.85 MB → 14.85 MB (+2.64 kB) | +0.02%
loot-core | 1 | 5.82 MB | 0%
api | 1 | 4.43 MB | 0%

<details>
<summary>View detailed bundle stats</summary>

#### desktop-client

**Total**
Files count | Total bundle size | % Changed
----------- | ----------------- | ---------
27 | 14.85 MB → 14.85 MB (+2.64 kB) | +0.02%

<details>
<summary>Changeset</summary>

File | Δ | Size
---- | - | ----
`src/components/reports/reports/Crossover.tsx` | 📈 +2.14 kB (+5.09%) | 42 kB → 44.14 kB
`src/components/reports/spreadsheets/crossover-spreadsheet.ts` | 📈 +382 B (+4.00%) | 9.33 kB → 9.7 kB
`src/components/reports/reports/CrossoverCard.tsx` | 📈 +128 B (+1.16%) | 10.74 kB → 10.86 kB
`src/components/reports/graphs/CrossoverGraph.tsx` | 📈 +9 B (+0.08%) | 10.43 kB → 10.44 kB
</details>

<details>
<summary>View detailed bundle breakdown</summary>
<div>

**Added**
No assets were added

**Removed**
No assets were removed

**Bigger**
Asset | File Size | % Changed
----- | --------- | ---------
static/js/ReportRouter.js | 1.16 MB → 1.16 MB (+2.64 kB) | +0.22%

**Smaller**
No assets were smaller

**Unchanged**
Asset | File Size | % Changed
----- | --------- | ---------
static/js/index.js | 9.54 MB | 0%
static/js/indexeddb-main-thread-worker-e59fee74.js | 12.94 kB | 0%
static/js/workbox-window.prod.es5.js | 5.64 kB | 0%
static/js/ca.js | 188.15 kB | 0%
static/js/da.js | 106.35 kB | 0%
static/js/de.js | 180.07 kB | 0%
static/js/en-GB.js | 7.18 kB | 0%
static/js/en.js | 170.37 kB | 0%
static/js/es.js | 174.55 kB | 0%
static/js/fr.js | 179.6 kB | 0%
static/js/it.js | 171.16 kB | 0%
static/js/nb-NO.js | 156.96 kB | 0%
static/js/nl.js | 106.37 kB | 0%
static/js/pl.js | 88.37 kB | 0%
static/js/pt-BR.js | 154.22 kB | 0%
static/js/th.js | 181.87 kB | 0%
static/js/uk.js | 214.74 kB | 0%
static/js/resize-observer.js | 18.37 kB | 0%
static/js/BackgroundImage.js | 120.54 kB | 0%
static/js/narrow.js | 637.68 kB | 0%
static/js/TransactionList.js | 106.22 kB | 0%
static/js/wide.js | 164.15 kB | 0%
static/js/AppliedFilters.js | 9.71 kB | 0%
static/js/usePayeeRuleCounts.js | 10.04 kB | 0%
static/js/useTransactionBatchActions.js | 13.23 kB | 0%
static/js/FormulaEditor.js | 1.04 MB | 0%
</div>
</details>

---

#### loot-core

**Total**
Files count | Total bundle size | % Changed
----------- | ----------------- | ---------
1 | 5.82 MB | 0%

<details>
<summary>View detailed bundle breakdown</summary>
<div>

**Added**
No assets were added

**Removed**
No assets were removed

**Bigger**
No assets were bigger

**Smaller**
No assets were smaller

**Unchanged**
Asset | File Size | % Changed
----- | --------- | ---------
kcab.worker.BwrdDDMW.js | 5.82 MB | 0%
</div>
</details>

---

#### api

**Total**
Files count | Total bundle size | % Changed
----------- | ----------------- | ---------
1 | 4.43 MB | 0%

<details>
<summary>View detailed bundle breakdown</summary>
<div>

**Added**
No assets were added

**Removed**
No assets were removed

**Bigger**
No assets were bigger

**Smaller**
No assets were smaller

**Unchanged**
Asset | File Size | % Changed
----- | --------- | ---------
bundle.api.js | 4.43 MB | 0%
</div>
</details>
</details>

### CODE REVIEW

**Comment by youngcw** [comment]
@math29 This has conficts that need resolved before I can review

**Comment by jonner** [comment]
@math29 do you plan to continue this PR? If not, I can try to rebase to latest master.

**Comment by math29** [comment]
> @math29 do you plan to continue this PR? If not, I can try to rebase to latest master.

Thank you for proposing. I rebased the PR :)

**Comment by math29** [comment]
Without Inflation Rate provided: 
image

With Inflation Rate provided:
image

File: upcoming-release-notes/6495.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+category: Features
```
Comment (youngcw) [inline upcoming-release-notes/6495.md]: This is an enhancement

File: packages/desktop-client/src/components/reports/spreadsheets/crossover-inflation.test.ts
Comment (youngcw) [inline packages/desktop-client/src/components/reports/spreadsheets/crossover-inflation.test.ts:1]: Please combine the test files.

File: packages/desktop-client/src/components/reports/spreadsheets/crossover-spreadsheet.ts [hunk 2]
```diff
@@ -1,3 +1,19 @@
+/**
+ * Crossover Point Spreadsheet Calculator
+ *
+ * This module provides calculations for determining the "crossover point" - the moment
+ * when passive investment income exceeds living expenses, indicating financial independence.
+ *
+ * Key features:
```
Comment (youngcw) [inline packages/desktop-client/src/components/reports/spreadsheets/crossover-spreadsheet.ts]: Please remove these added comments.  They are mostly just clutter

**Comment by jonner** [comment]
It looks like you added a tooltip and a dotted line for non-inflation-adjusted expenses. But the 'target income' adjustment setting also generates a dotted line for the adjusted target income. So if both of those settings are used, there will be two dotted lines on the graph. I worry a little bit that things are starting to get too crowded and may require a bigger overhaul of the UI to simplify things.

**Comment by math29** [comment]
> It looks like you added a tooltip and a dotted line for non-inflation-adjusted expenses. But the 'target income' adjustment setting also generates a dotted line for the adjusted target income. So if both of those settings are used, there will be two dotted lines on the graph. I worry a little bit that things are starting to get too crowded and may require a bigger overhaul of the UI to simplify things.

Taking your comment into consideration, I think best approach would be to keep this new monthly inflation tooltip. We should have 2 lines displayed:
- 1 plain line = target income with adjustement + inflation
- 1 dotted line = target income without any adjustement nor inflation

What do you think?

**Comment by jonner** [comment]
That sounds reasonable to me.

**Comment by math29** [comment]
> That sounds reasonable to me.

Adapted the PR accordingly
Also I am proposing to change the color of adjustedExpenses (with inflation) to dashed blue as it is weird to have 2 lines with same color and/or initial expenses line with dashed red line. wdyt? 
image

image

**Comment by youngcw** [comment]
> > That sounds reasonable to me.
> 
> Adapted the PR accordingly Also I am proposing to change the color of adjustedExpenses (with inflation) to dashed blue as it is weird to have 2 lines with same color and/or initial expenses line with dashed red line. wdyt? image
> 
> image

I think the lines used for the calculation should be the solid lines.  Dashed lines should be a "incase you were curious" data point.

**Comment by jonner** [comment]
> > image
> 
> I think the lines used for the calculation should be the solid lines. Dashed lines should be a "incase you were curious" data point.

I agree with this. I also noticed that in the screenshot above, the value in the tooltip for "Target Monthly Expenses" is lower than "Target monthly Expenses (no inflation)". I think you got the tooltip labels swapped?

**Comment by math29** [comment]
> At least 2 approving reviews are required by reviewers with write access.

Indeed, it was swapped by mistake. Now solved. Also moved line to solid line (until now, adjusted line was dashed red). Here after a screenshot of the new rendering VS before when it was dashed line.
image
VS (before)
image


And swapped back labels 
image

**Comment by youngcw** [comment]
This is weird.  Leave the red line as the one used for calculations and make it adapt to inflation.  If wanted make the non-inflated line a dashed red line or don't include it.

Having the colors change randomly is strange.  It also is weird when there is no inflation adjustment and the line is suddenly blue.
image
